### DISCUSSION TITLE
Elsa workflow how to get Workflow Instance Id

### DISCUSSION THREAD

**Discussion by albourm** [elsa-workflows/elsa-core#5240] [Q&A]
Hello there,

I am using Elsa Workflow V3 to Create Workflow using The Designer so i want to get the Workflow instance id to be passed to other activity  using JS Code in the Designer.

So, How to get the workflow instance id

Thanks.

**Comment by glime-ai** [comment]
To get the workflow instance ID in Elsa Workflow V3 using JavaScript code in the designer, you can use the built-in JavaScript functions provided by Elsa. However, the exact method to retrieve the workflow instance ID may vary depending on the context in which you are trying to access it.

If you are trying to access the workflow instance ID within an activity's JavaScript expression, you can typically use the `workflowInstance` object that is available in the JavaScript execution context. For example, you might use the following JavaScript code:

```javascript
// Get the workflow instance ID
var instanceId = workflowInstance.Id;
```

Please note that the above code is a general example. The actual object and property names may differ based on the version of Elsa Workflow you are using and the context in which you are executing the JavaScript code.

If you are unable to retrieve the workflow instance ID using the method described above, or if you encounter any issues, please provide more details about where and how you are trying to access the workflow instance ID so that I can assist you further.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by albourm** [comment]
I want to get the Actual Workflow instance id not for the specific activity

**Comment by glime-ai** [comment]
To get the workflow instance ID in Elsa Workflow V3 using JavaScript code in the designer, you can use the built-in JavaScript functions provided by Elsa. However, the exact method to retrieve the workflow instance ID may vary depending on the context in which you are executing the JavaScript code.

If you are trying to access the workflow instance ID within an activity, you can typically use the `workflowInstance` object that is available in the JavaScript execution context. Here's an example of how you might access the workflow instance ID:

```javascript
// Access the workflow instance ID
var instanceId = workflowInstance.Id;
```

Please note that the above code is a general example and may need to be adjusted based on the specific context and version of Elsa Workflow you are using.

If you are unable to retrieve the workflow instance ID using the method described above, or if you encounter any issues, please provide more details about where and how you are trying to access the workflow instance ID so that I can assist you further.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by SydoxX** [comment]
```js
`getWorkflowInstanceId()`
```
does the job for me.
